ext/standard/basic_functions.c

```diff
@@ -1214,18 +1216,32 @@ PHP_FUNCTION(flush)
 /* {{{ Delay for a given number of seconds */
 PHP_FUNCTION(sleep)
 {
-	zend_long num;
+	double seconds;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_LONG(num)
+		Z_PARAM_DOUBLE(seconds)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (num < 0) {
+	if (seconds < 0) {
 		zend_argument_value_error(1, "must be greater than or equal to 0");
 		RETURN_THROWS();
 	}
 
-	RETURN_LONG(php_sleep((unsigned int)num));
+#if HAVE_NANOSLEEP
+	struct timespec php_req, php_rem;
+
+	php_req.tv_sec = (time_t) ((uint64_t)(seconds * NANOS_IN_SEC) / NANOS_IN_SEC);
+	php_req.tv_nsec = (long) ((uint64_t)(seconds * NANOS_IN_SEC) % NANOS_IN_SEC);
+	if (!nanosleep(&php_req, &php_rem)) {
+		RETURN_DOUBLE(0);
+	} else if (errno == EINTR) {
+		RETURN_DOUBLE((double)php_rem.tv_sec + (double)php_rem.tv_nsec / NANOS_IN_SEC);
+	}
+#else
+	RETURN_DOUBLE((double) php_sleep((unsigned int)seconds));
+#endif
+
+	RETURN_FALSE;
```

Comment:
```
php > echo (new ReflectionFunction('sleep'));
Function [ <internal:standard> function sleep ] {

  - Parameters [1] {
    Parameter #0 [ <required> int $seconds ]
  }
  - Return [ int ]
}
```

```php
ext/standard/basic_functions.stub.php
272:function sleep(int $seconds): int {}
```

The function signature should be changed back to int|false in the stub file in ext/standard and the regenerated _arginfo files should be committed with the change, if we're doing this. (to avoid opcache bugs)

Reply:
php stub change reverted

Reply:
@mvorisek update the `.stub.php` file and then run `build/gen_stub.php` to regenerate the `_arginfo.h`.

Reply:
@moliata thanks, done